tutorials/math/bit-manipulation.md

```diff
@@ -268,4 +290,55 @@ while (n > 0) {
     // which is same as n /= 2
     n >>= 1;
 }
+```
+
+### Two's Compliment and Negative Numbers
+
+Computers typically store integers in two’s complement representation. A positive number is represented as itself while a negative number is represented as the two’s complement of it’s absolute value (with a 1 in its sign bit to indicate that a negative value). 
+
+Let’s look at the `4-bit` integer $-3$ as an example. _If it’s a 4-bit number, we have one bit for the sign and three bits for the value_. We want the complement with respect to $2^3$, which is $8$. 
+
+The complement of $3$ (the absolute value of $-3$) with respect to $8$ is $5$. Binary value of $5$ is $0101_2$. Therefore, $-3$ in binary as a 4-bit number is $1101_2$, with the first bit being the sign bit. 
+
+In other words, the binary representation of $-K$ (negative K) as a N-bit number is $concat(1, 2^N-1 - K)$
+
+Another way to look at this is that we invert the bits in the positive representation and then add $1$. $3$ is $0011_2$ in binary. 
+
+_Flip the bits_ to get 100, add $1$ to get $101_2$, then prepend the sign bit $1$ to get $1101_2$.
+
+```
+Positive  Values  Negative  Values
+--------  ------  --------  ------
+7         0 111   -1        1 111
+6         0 110   -2        1 110
+5         0 101   -3        1 101
+4         0 100   -4        1 100
+3         0 011   -5        1 011
+2         0 010   -6        1 010
+1         0 001   -7        1 001
+0         0 000 
+```
+
+### Common Bit Tasks
+
+### Get Bit
+It shifts $1$ over by _i_ bits, creating a value that looks like $0100_2$ $(2^i)$. By performing and AND $ (\&) $ with num, we clear all bits other than the bit at bit _i_.
```

Comment:
`$ (\&) $` is broken. Check in your local.

Reply:
Changed and pushed it again.